flo-runner/src/main/java/com/spotify/flo/context/FloRunner.java

```diff
@@ -234,21 +248,28 @@ public static String randomAlphaNumeric(int count) {
 
   public static class Result<T> {
     private final Future<T> future;
+    private final Consumer<Integer> exiter;
 
-    Result(Future<T> future) {
+    Result(Future<T> future, Consumer<Integer> exiter) {
       this.future = future;
+      this.exiter = (Integer exitCode) -> {
+        exiter.accept(exitCode);
```

Comment:
We need to catch and ignore exception here. If javaslang is already a dependency we can use Try.of.